### PR TITLE
Deployment

### LINKED ISSUE #4: Set up CubeJS in production mode

### PR DESCRIPTION
**Pull request by rathorevaibhav**
Fixes #4 

Includes:
1. Docker file and docker ignore that will be used for ECR docker image.
2. Github workflow to create an image and push to ECR and deploy on ECS.
3. Updates to .env.example and documentation for deployment and authentication
4. Issue & PR templates added

### CODE REVIEW

File: docs/ENV.md [hunk 1]
```diff
@@ -37,3 +40,31 @@ Port on which you want the Cube.js API to be exposed.
 
 #### `APP_DASHBOARD_PORT`
 Port on which you want the Cube.js Dashboard App to be created using the Developer Playground.
+
```
Comment (deepansh96) [inline docs/ENV.md:43]: In the `.env.example` I saw that you changed `APP_API_PORT` to `PORT`, but that change is not reflected here

File: docs/ENV.md [hunk 2]
```diff
@@ -37,3 +40,31 @@ Port on which you want the Cube.js API to be exposed.
...
+### Redis
+#### `CUBEJS_REDIS_URL`
+Hostname of your Redis instance.
+
+#### `CUBEJS_REDIS_PASSWORD`
+Password of your Redis instance.
+
+#### `CUBEJS_REDIS_TLS`
+To set TLS authentication between the app and Redis server.
```
Comment (deepansh96) [inline docs/ENV.md]: Can you just add a word saying that it's boolean?
was not super clear to me at first glance

File: docs/ENV.md [hunk 3]
```diff
@@ -37,3 +40,31 @@ Port on which you want the Cube.js API to be exposed.
...
+
+### Authentication
+Before setting these variables, make sure you have gone through the [Authentication](./AUTHENTICATION.md) guide.
+
+#### `CUBEJS_JWK_URL`
+This is the url to fetch the JSON wek keys details. Replace the `<AUTH0-SUBDOMAIN>` by the domain name for your account. The domain can be determined from `Your app > Settings > Domain`.
```
Comment (deepansh96) [inline docs/ENV.md]:
```suggestion
This is the url to fetch the JSON web keys details. Replace the `<AUTH0-SUBDOMAIN>` by the domain name for your account. The domain can be determined from `Your app > Settings > Domain`.
```

File: docs/DEPLOYMENT.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+10. AWS Identity and Access Management (IAM)
+11. AWS Relational Database Service (RDS)
+
+## Staging
+
+### Continuous Delivery process
+
+An overview of how the continuous delivery cycle works in Plio with GitHub action and Amazon ECR & ECS.
+
+![Overview of Continuous Delivery process](images/aws-gh-cd.png)
```
Comment (deepansh96) [inline docs/DEPLOYMENT.md:29]: Can't see this image!

File: docs/DEPLOYMENT.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+Follow the steps below to set up the staging environment on AWS.
+
+1. Login to your AWS console.
+
+2. Go to VPC. (skip this step if you've already created a VPC when setting up the frontend or analytics repository)
```
Comment (deepansh96) [inline docs/DEPLOYMENT.md]:
```suggestion
2. Go to VPC. (skip this step if you've already created a VPC when setting up the frontend or backend repository)
```

File: docs/DEPLOYMENT.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    7. Use the `.env.example` file to set all the required environment variables for your container in the `Environment Variables` section.
+       1. Set all `CUBEJS_DB_*` variables to your AWS RDS setup. `CUBEJS_DB_TYPE` should be postgres.
+       2. For production, set `CUBEJS_DEV_MODE` to false.
+       3. Ignore the variables under `local web server`.
+       4. Leave the variables `CUBEJS_REDIS_URL` and `CUBEJS_REDIS_PASSWORD` as empty for now. We'll fill them up later.
+       5. Ignore the variables under `local web server`.
+    8.  Save the container definition and the task definition.
+    9.  You will see the new task definition within the list of all task definitions.
+
+8. Under ECS, go to `Clusters` and create a new cluster with the name `plio-staging-cluster`. (skip this step if you've already created a Cluster when setting up the frontend repository)
```
Comment (deepansh96) [inline docs/DEPLOYMENT.md]:
```suggestion
8. Under ECS, go to `Clusters` and create a new cluster with the name `plio-staging-cluster`. (skip this step if you've already created a Cluster when setting up the frontend/backend repository)
```

File: docs/DEPLOYMENT.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+      1. Go to the [ElastiCache dashboard](https://console.aws.amazon.com/elasticache/home). Select the region and click on `Create` button.
+      2. Select `Redis` as your Cluster engine.
+      3. Select `Cluster mode disabled`. More details on cluster mode [here](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/Replication.Redis-RedisCluster.html).
+      4. Choose `Amazon Cloud` as your location.
+      5. Set the name of your cluster as `redis-plio-analytics-staging`.
+      6. Choose the latest `Engine version compatibility`.
+      7. Enter `6379` as the `Port`.
+      8. No need to change anything for `Parameter Group`.
+      9. Choose your `Node Type`. Plio uses `cache.t2.micro`. More details [here](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/nodes-select-size.html#CacheNodes.SelectSize).
+      10. For `Number of replicas`, enter `2`.
```
Comment (deepansh96) [inline docs/DEPLOYMENT.md]:
```suggestion
      10. For `Number of replicas`, enter `0`. Note: Each replica node will carry it's own cost. More on pricing [here](https://aws.amazon.com/elasticache/pricing/) and [here](https://www.reddit.com/r/aws/comments/cojaq6/questions_about_elasticache_pricing/)
```

File: docs/AUTHENTICATION.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
+## Authentication
+Plio Analytics uses Auth0 authentication to manage access tokens and valid API requests from the client app, which in Plio's ecosystem is the frontend app.
```
Comment (deepansh96) [inline docs/AUTHENTICATION.md]:
```suggestion
Plio Analytics uses Auth0 authentication to manage access tokens and valid API requests from a client app.
```
I don't think it's needed. Any client app can connect to this analytics service right?
Just an opinion though!

File: docs/ENV.md [hunk 9]
```diff
@@ -37,3 +40,31 @@ Port on which you want the Cube.js API to be exposed.
...
+
+#### `CUBEJS_JWT_AUDIENCE`
+This is the unique identifier that was set when configuring the API in Auth0. Should be the url for your app.
+
+#### `CUBEJS_JWT_ISSUER`
+JWT issuer should be the domain for your Auth0 app. Replace the `<AUTH0-SUBDOMAIN>` by the domain name for your account. The domain can be determined from `Your app > Settings > Domain`.
+
+#### `CUBEJS_JWT_ALGS`
+This is to remain RS256 to decode the access token generated by same algorithm.
```
Comment (deepansh96) [inline docs/ENV.md]:
```suggestion
This should be set to RS256 to decode the access token generated by same algorithm.
```

File: docs/AUTHENTICATION.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+```sh
+CUBEJS_JWK_URL=https://<AUTH0-SUBDOMAIN>.auth0.com/.well-known/jwks.json
+CUBEJS_JWT_AUDIENCE=<APPLICATION_URL>
+CUBEJS_JWT_ISSUER=https://<AUTH0-SUBDOMAIN>.auth0.com/
+CUBEJS_JWT_ALGS=RS256
+CUBEJS_JWT_CLAIMS_NAMESPACE=<CLAIMS_NAMESPACE>
+```
+
+#### `CUBEJS_JWK_URL`
+This is the url to fetch the JSON wek keys details. Replace the `<AUTH0-SUBDOMAIN>` by the domain name for your account. The domain can be determined from `Your app > Settings > Domain`.
```
Comment (deepansh96) [inline docs/AUTHENTICATION.md]:
```suggestion
This is the url to fetch the JSON web keys details. Replace the `<AUTH0-SUBDOMAIN>` by the domain name for your account. The domain can be determined from `Your app > Settings > Domain`.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
+<!--
+  Thanks for submitting a pull request!
+  We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request.
+
+  Before submitting a pull request, please make sure the following is done:
+
+  1. Fork [the repository](https://github.com/avantifellows/plio-backend) and create your branch from `master`.
```
Comment (deepansh96) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
  1. Fork [the repository](https://github.com/avantifellows/plio-analytics) and create your branch from `master`.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+  2. Run the installation steps from the project's [README.md](https://github.com/avantifellows/plio-backend#readme).
```
Comment (deepansh96) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
  2. Run the installation steps from the project's [README.md](https://github.com/avantifellows/plio-analytics#readme).
```

File: .env.example [hunk 13]
```diff
@@ -10,6 +10,22 @@ CUBEJS_DB_TYPE=postgres
 CUBEJS_API_SECRET= # random 32-digit base64 string - generate from https://generate.plus/en/base64
 CUBEJS_DEV_MODE=true
 
+PORT=4000
+
 # local web server
-APP_API_PORT=4000
```
Comment (deepansh96) [inline .env.example:14]: Why this change?

File: .env.example [hunk 14]
```diff
@@ -10,6 +10,22 @@ CUBEJS_DB_TYPE=postgres
...
 APP_DASHBOARD_PORT=3000
+
+# redis configuration
+CUBEJS_REDIS_URL=
+CUBEJS_REDIS_PASSWORD=
+CUBEJS_REDIS_TLS=true
+
+# Set to true so a Cube.js instance acts as a refresh worker
```
Comment (deepansh96) [inline .env.example]:
```suggestion
# Set to true to make a Cube.js instance act as a refresh worker
```

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: .env.example [hunk 15]
```diff
@@ -10,6 +10,22 @@ CUBEJS_DB_TYPE=postgres
 CUBEJS_API_SECRET= # random 32-digit base64 string - generate from https://generate.plus/en/base64
 CUBEJS_DEV_MODE=true
 
+PORT=4000
```
Comment (dalmia) [inline .env.example:13]: will need to update docker-compose right?

File: .github/workflows/deploy_to_ecs_staging.yml [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+
+      # Uses the ECS service for analytics to create a new task from the new task definition. The new task is created while the previous task is already running. This happens based on the minimum healthy percentage and maximum percentage configured during service creation.
+      # After service completion and stability, it removes the previous task from older task definition.
+      - name: Deploy Amazon ECS task definition
+        uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+        with:
+          task-definition: ${{ steps.task-def.outputs.task-definition }}
+          service: plio-analytics-staging
+          cluster: plio-staging-cluster
+          wait-for-service-stability: true
```
Comment (dalmia) [inline .github/workflows/deploy_to_ecs_staging.yml:74]: any reason we are not using this in the other repos too?

File: docs/ENV.md [hunk 17]
```diff
@@ -1,6 +1,6 @@
 ## Environment variables
 
-This guide explains all the available configurations in your `.env` file so that you can configure your environments accordingly.
+This guide explains all the available configurations in your `.env` file so that you can configure your environments accordingly. For more details, you can also check official documentation from Cube.js on [Environment Variables](https://cube.dev/docs/reference/environment-variables).
```
Comment (dalmia) [inline docs/ENV.md:3]: the link talks about this env variable:
```
CUBEJS_APP | An application ID used to uniquely identify the Cube.js deployment. Can be different for multitenant setups. Defaults to cubejs
```
Wouldn't we need this for our multitenancy part?

File: docs/ENV.md [hunk 18]
```diff
@@ -10,6 +10,9 @@ Debug mode on or off. Possible values are `True` and `False`. Do not set to true
 Used by CubeJS. It should be a secret value and shouldn't be added to git.
 You can create a 32-digit secret key from this website: https://generate.plus/en/base64
 
+#### `CUBEJS_SCHEDULED_REFRESH_TIMER`
```
Comment (dalmia) [inline docs/ENV.md:13]: are we using pre-aggregations?

Comment (dalmia) [reply docs/ENV.md:13]: If this is for refreshing in-memory cache, what should be the value - just `true` or some set number of seconds - as a nice default?

Comment (dalmia) [reply docs/ENV.md:13]: how are we triggering scheduled refreshes? this is the recommended process: https://cube.dev/docs/config#options-reference-scheduled-refresh-timer

File: .env.example [hunk 19]
```diff
@@ -10,6 +10,22 @@ CUBEJS_DB_TYPE=postgres
...
+
 # local web server
-APP_API_PORT=4000
 APP_DASHBOARD_PORT=3000
+
+# redis configuration
+CUBEJS_REDIS_URL='redis://host.docker.internal:6379'
+CUBEJS_REDIS_PASSWORD=
```
Comment (dalmia) [inline .env.example:20]: is this the right default?

File: .env.example [hunk 20]
```diff
@@ -10,6 +10,22 @@ CUBEJS_DB_TYPE=postgres
...
+CUBEJS_REDIS_TLS=true
```
Comment (dalmia) [inline .env.example]: should this be true for development?

File: .env.example [hunk 21]
```diff
@@ -10,6 +10,22 @@ CUBEJS_DB_TYPE=postgres
...
 CUBEJS_DEV_MODE=true
 
+PORT=4000
+
 # local web server
-APP_API_PORT=4000
 APP_DASHBOARD_PORT=3000
+
+# redis configuration
+CUBEJS_REDIS_URL='redis://host.docker.internal:6379'
```
Comment (dalmia) [inline .env.example:19]: for local - this is not even needed right?

File: .env.example [hunk 22]
```diff
@@ -10,6 +10,22 @@ CUBEJS_DB_TYPE=postgres
...
+CUBEJS_REDIS_PASSWORD=
+CUBEJS_REDIS_TLS=true
+
+# Set to true so a Cube.js instance acts as a refresh worker
+CUBEJS_SCHEDULED_REFRESH_TIMER=false
+
+# authentication in production mode
+CUBEJS_JWK_URL=
```
Comment (dalmia) [inline .env.example:27]: don't we need `CUBEJS_JWT_SUBJECT` and `CUBEJS_JWT_KEY`?

File: .env.example [hunk 23]
```diff
@@ -10,6 +10,22 @@ CUBEJS_DB_TYPE=postgres
 CUBEJS_API_SECRET= # random 32-digit base64 string - generate from https://generate.plus/en/base64
 CUBEJS_DEV_MODE=true
 
+PORT=4000
+
```
Comment (dalmia) [inline .env.example:14]: wouldn't we need more DB env variables like: `CUBEJS_DB_SCHEMA` (multi-tenancy), `CUBEJS_DB_SSL`, `CUBEJS_AWS_KEY`? 
Are we using pre-aggregations? are we using Redshift or RDS?

File: docs/AUTHENTICATION.md [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+### Configurating Auth0
+1. Sign up or log into your [Auth0 account](https://auth0.com/).
+2. From the left navigation, go to APIs and create a new API.
+   1. Name your API.
+   2. Set the identifier to your app url. Like `https://myapp.com/`. This must be a unique identifier and not to be reused in other APIs.
+   3. Signing Algorithm should be `RS256`.
+3. From the left navigation, go to Applications.
+   1. You will see an application was created automatically with your api name and `(Test application)`.
+   2. Select this app and go to settings tab. Rename by removing the (Test Application) part and save.
+4. Go to Quick Start tab and copy the example code for "Getting an access token for your API". Run the command in your shell. It should return a valid response with access token, expiry time and token type.
```
Comment (dalmia) [inline docs/AUTHENTICATION.md]: what is to be done with the access token and other values? is this only for testing?

File: docs/DEPLOYMENT.md [hunk 25]
```diff
@@ -0,0 +1,215 @@
...
+8. AWS Elastic Load Balancer (ELB)
+9. AWS Elastic IPs
+10. AWS Identity and Access Management (IAM)
+11. AWS Relational Database Service (RDS)
+
+## Staging
+
+### Continuous Delivery process
+
+An overview of how the continuous delivery cycle works in Plio with GitHub action and Amazon ECR & ECS.
```
Comment (dalmia) [inline docs/DEPLOYMENT.md:27]: Have we set it up for multi-tenancy?

Comment (dalmia) [reply docs/DEPLOYMENT.md:27]: https://cube.dev/docs/multitenancy-setup#top

File: docs/AUTHENTICATION.md (hunk 8 above)
Comment (dalmia) [inline docs/AUTHENTICATION.md]: Are we using Security context?

File: docs/AUTHENTICATION.md [hunk 26]
```diff
@@ -0,0 +1,43 @@
+## Authentication
```
Comment (dalmia) [inline docs/AUTHENTICATION.md:1]: need to set refresh timer to 20 days or something.

File: docs/DEPLOYMENT.md (hunk 7 above)
Comment (deepansh96) [inline docs/DEPLOYMENT.md]: REMINDER: delete and create a new cluster on ElastiCache

File: .env.example (hunk 13 above)
Comment (rathorevaibhav) [reply .env.example:14]: there is a default cubejs variable `PORT` that is used by cubejs to switch the port. I think we can use the same for docker-compose as well.

File: .env.example (hunk 15 above)
Comment (rathorevaibhav) [reply .env.example:13]: yep done.

File: .github/workflows/deploy_to_ecs_staging.yml (hunk 16 above)
Comment (rathorevaibhav) [reply .github/workflows/deploy_to_ecs_staging.yml:74]: didn't get your question, @dalmia 

This code is same as in other repos except the service/task name etc.

File: docs/ENV.md (hunk 17 above)
Comment (rathorevaibhav) [reply docs/ENV.md:3]: I believe this is for multitenant set up of cubejs.

File: .env.example (hunk 19 above)
Comment (rathorevaibhav) [reply .env.example:20]: yep. by default redis docker doesn't come with a password. Elasticache also doesn't come with a password by default. However Elasticache redis connection can only be made within the same VPC.

File: .env.example (hunk 21 above)
Comment (rathorevaibhav) [reply .env.example:19]: yeah redis is not needed for local setup. Cubejs switches to memory when dev mode is enabled.

File: .env.example (hunk 22 above)
Comment (rathorevaibhav) [reply .env.example:27]: Not needed for Auth0 access token verification.

File: docs/AUTHENTICATION.md (hunk 24 above)
Comment (rathorevaibhav) [reply docs/AUTHENTICATION.md]: This is only for testing. i'll update the step.

File: .env.example (hunk 20 above)
Comment (rathorevaibhav) [reply .env.example]: yeah it needs to be false for development. I've updated the .env.example

File: docs/AUTHENTICATION.md (hunk 8 above)
Comment (rathorevaibhav) [reply docs/AUTHENTICATION.md]: Right now we'll proceed without security context as we have machine-to-machine API calls between frontend and analytics.

The ACL checks will be handled by frontend/backend repos.

File: docs/DEPLOYMENT.md (hunk 25 above)
Comment (rathorevaibhav) [reply docs/DEPLOYMENT.md:27]: Working on it.

File: docs/AUTHENTICATION.md [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+Plio Analytics uses Auth0 authentication to manage access tokens and valid API requests from the client app, which in Plio's ecosystem is the frontend app.
+
```
Comment (dalmia) [inline docs/AUTHENTICATION.md:3]: Are we using auth0.js or auth sdk spa?

File: docs/AUTHENTICATION.md (hunk 8 above)
Comment (dalmia) [inline docs/AUTHENTICATION.md]: Screenshot 2021-05-12 at 12 56 49 PM
Here the key mentioned is for connecting cubejs to the JWK - and the JWT is the key for each client of cubejs (like our plio backend) right?

File: .env.example (hunk 22 above)
Comment (dalmia) [reply .env.example:27]: are we not using subject claim because it is a machine-machine thing?

File: docs/DEPLOYMENT.md [hunk 28]
```diff
@@ -0,0 +1,215 @@
+## Deployment
+This document covers the steps on setting up this repository on various cloud hosting providers.
```
Comment (dalmia) [inline docs/DEPLOYMENT.md:2]: Do we have the required number of concurrent connections supported?
this is dependent on multi-tenant count - should probably also update multi-tenancy to say that one needs to ensure that this requirement for Redis is met when setting up a new tenant?

File: docs/DEPLOYMENT.md [hunk 29]
```diff
@@ -0,0 +1,215 @@
...
+
```
Comment (dalmia) [inline docs/DEPLOYMENT.md:3]: How are we enabling SSL/TLS for cube? https://cube.dev/docs/deployment/production-checklist#enable-https

File: docs/DEPLOYMENT.md [hunk 30]
```diff
@@ -0,0 +1,215 @@
...
+    3. Set the task memory and task CPU based on your needs. Good defaults to use are: 2 GB for memory and 0.5 vCPU.
+    4. Create a new container with name `plio-analytics-staging`.
+    5. In the image field, you can just type in `image_arn`. This is not a valid entry and just a placeholder for now as it'll be replaced by the actual image ARN once the GitHub workflow triggers.
+    6. Enter port `80` in the port mapping field.
+    7. Use the `.env.example` file to set all the required environment variables for your container in the `Environment Variables` section.
+       1. Set all `CUBEJS_DB_*` variables to your AWS RDS setup. `CUBEJS_DB_TYPE` should be postgres.
+       2. For production, set `CUBEJS_DEV_MODE` to false.
+       3. Ignore the variables under `local web server`.
+       4. Leave the variables `CUBEJS_REDIS_URL` and `CUBEJS_REDIS_PASSWORD` as empty for now. We'll fill them up later.
+       5. Ignore the variables under `local web server`.
```
Comment (dalmia) [inline docs/DEPLOYMENT.md]: this seems the same as sub-point 3 right?

File: docs/DEPLOYMENT.md [hunk 31]
```diff
@@ -0,0 +1,215 @@
...
+    4. Select the running task definition in the `Task Definition` column.
+    5. Click `Create new revision` button on the top and select the existing container with the name: `plio-analytics-staging`.
+    6. In the new window, scroll down to `Environment Variables` section.
+    7. Update the `CUBEJS_REDIS_URL` variable that were copied before. By default, AWS Elasticache has no password. In case you had configured it, make sure to update `CUBEJS_REDIS_PASSWORD` variable as well.
+    8. Click the `Update` button to update the container definition.
+    9. Click the `Create` button to create a new revision of the task definition.
+    10. Navigate back to the service `plio-analytics-staging`.
+    11. Click `Update` on top right.
+    12. Update the revision of the newly created task definition and wait for it to get initiated.
+    13. Wait for some time. The older task will automatically stop once the new task reaches a stable state.
```
Comment (dalmia) [inline docs/DEPLOYMENT.md:177]: Does this happen automatically? I remember having to manually kill the older tasks - maybe I didn't wait long enough?

File: docs/AUTHENTICATION.md [hunk 32]
```diff
@@ -15,6 +15,7 @@ For more details on Auth0 configuration, check out the [official documentation f
    1. You will see an application was created automatically with your api name and `(Test application)`.
    2. Select this app and go to settings tab. Rename by removing the (Test Application) part and save.
 4. To test your set up, go to Quick Start tab and copy the example code for "Getting an access token for your API". Run the command in your shell. It should return a valid response with access token, expiry time and token type.
+5. Make sure you have updated the Plio backend repository [environment variables](https://github.com/avantifellows/blob/master/docs/ENV.md#auth0-for-plio-analytics) with the Auth0 app settings.
```
Comment (dalmia) [inline docs/AUTHENTICATION.md]: the link seems to be invalid (the repo name is missing)

File: docs/AUTHENTICATION.md (hunk 27 above)
Comment (dalmia) [reply docs/AUTHENTICATION.md:3]: Or we are ignoring it for now as it seems to be important for custom claims? https://cube.dev/docs/security/jwt/auth0#create-an-application-custom-claims

**Comment by dalmia** [comment]
We should ideally pin all our doubts and their answers in one place on Notion for future reference.

File: docs/DEPLOYMENT.md (hunk 31 above)
Comment (rathorevaibhav) [reply docs/DEPLOYMENT.md:177]: yeah it happens automatically. I was also stopping tasks manually but later figured out that it happens automatically. just need to wait for some time.

File: .github/workflows/deploy_to_ecs_staging.yml (hunk 16 above)
Comment (dalmia) [reply .github/workflows/deploy_to_ecs_staging.yml:74]: My bad. I thought `wait-for-service-stability` was missing in other repos.

File: schema/Plio.js [hunk 33]
```diff
@@ -1,5 +1,8 @@
+const { securityContext: { tenantSchema } } = COMPILE_CONTEXT;
+// const { organizationTenantSchema } = COMPILE_CONTEXT;
+
 cube(`Plio`, {
-  sql: `SELECT * FROM public.plio`,
+  sql: `SELECT * FROM ${tenantSchema}.plio`,
```
Comment (dalmia) [inline schema/Plio.js:5]: we need to do this for the User.js file too right?

File: cube.js [hunk 34]
```diff
@@ -0,0 +1,17 @@
...
+    contextToAppId: ({ securityContext }) => `CUBEJS_APP_${securityContext.tenantSchema}`,
+    checkAuth: (req, token) => {
+        const tenantSchema = req.headers['organization']
+        req.securityContext = { tenantSchema }
+    },
+    extendContext: async (req) => {
+        return {
+            'organizationTenantSchema': 'abcd',
+        };
+    },
```
Comment (dalmia) [inline cube.js]: do we need this?

Comment (rathorevaibhav) [reply cube.js]: not needed. I missed pushing the latest commit.

File: .env.example (hunk 22 above)
Comment (rathorevaibhav) [reply .env.example:27]: yep, that's right. it's needed when we're doing user-level calls.

File: .env.example (hunk 23 above)
Comment (rathorevaibhav) [reply .env.example:14]: I think we can live without these keys for now.

The DB_SCHEMA doesn't have much documentation. Maybe it can help solving the issue with multitenancy that we're facing. Will look into this.

File: docs/DEPLOYMENT.md (hunk 7 above)
Comment (rathorevaibhav) [reply docs/DEPLOYMENT.md]: Done.

File: docs/AUTHENTICATION.md (hunk 27 above)
Comment (rathorevaibhav) [reply docs/AUTHENTICATION.md:3]: Yeah we don't need custom claims as of now. We're just relying on Auth0 for access token and auth verification at cubejs.
